org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/serialization/SerializationUtil.java

```diff
@@ -48,7 +48,9 @@ public static void fillIdToEObjectMap(Resource resource, List<EObject> map) {
 	}
 
 	public static void fillIdToEObjectMap(EObject eObject, List<EObject> map) {
-		map.add(eObject);
+		if (eObject.eContainingFeature() == null || !eObject.eContainingFeature().isTransient()) {
+			map.add(eObject);
+		}
```

Comment:
Children of transient object become transient automatically. They are after all bound to the lifecycle of their parent.